controller/src/main/java/org/jboss/as/controller/persistence/FilePersistenceUtils.java

```diff
@@ -77,9 +77,13 @@ static void copyFile(final File file, final File backup) throws IOException {
         Files.copy(file.toPath(), backup.toPath(), StandardCopyOption.COPY_ATTRIBUTES, StandardCopyOption.REPLACE_EXISTING);
     }
 
-    static void rename(File file, File to) throws IOException {
-        if (!file.renameTo(to) && file.exists()) {
-            copyFile(file, to);
+    static boolean rename(Path from, Path to) {
+        try {
+            Files.move(from, to);
+            return true;
+        } catch (IOException e) {
+            ControllerLogger.ROOT_LOGGER.cannotRename(e, from, to);
+            return false;
```

Comment:
Fair point; I just took an existing unused method and changed it's internals. No reason not to give it a new name as well.

I definitely really want a forced move. If someone has a different use case they can add a param. But, there's no reason I can't move this method to the only class that uses it which removes any question about whether its API is correct.